Intermittent failure of MyBatisConsumerTest

https://github.com/apache/camel-quarkus/actions/runs/4694358152/jobs/8323072795
```
[INFO] Running org.apache.camel.quarkus.component.mybatis.it.MyBatisConsumerTest
[INFO] H2 database started in TCP server mode; server status: TCP server running at tcp://10.1.0.16:9092 (only local connections)
2023-04-14 01:05:43,041 INFO  [org.apa.cam.qua.cor.CamelBootstrapRecorder] (main) Bootstrap runtime: org.apache.camel.quarkus.main.CamelMainRuntime
2023-04-14 01:05:43,046 INFO  [org.apa.cam.mai.MainSupport] (main) Apache Camel (Main) 4.0.0-M2 is starting
2023-04-14 01:05:43,271 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main) Apache Camel 4.0.0-M2 (camel-1) is starting
2023-04-14 01:05:43,287 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main) Routes startup (total:9 started:8 disabled:1)
2023-04-14 01:05:43,287 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main)     Started route1 (direct://selectOne)
2023-04-14 01:05:43,288 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main)     Started route2 (direct://selectList)
2023-04-14 01:05:43,288 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main)     Started route3 (direct://insertOne)
2023-04-14 01:05:43,289 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main)     Started route4 (direct://insertList)
2023-04-14 01:05:43,290 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main)     Started route5 (direct://deleteOne)
2023-04-14 01:05:43,290 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main)     Started route6 (direct://deleteList)
2023-04-14 01:05:43,290 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main)     Started route7 (direct://updateOne)
2023-04-14 01:05:43,291 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main)     Started route8 (direct://updateList)
2023-04-14 01:05:43,292 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main)     Disabled mybatis-consumer (mybatis://selectUnprocessedAccounts)
2023-04-14 01:05:43,293 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main) Apache Camel 4.0.0-M2 (camel-1) started in 109ms (build:0ms init:89ms start:20ms)
2023-04-14 01:05:43,440 INFO  [io.quarkus] (main) camel-quarkus-integration-test-mybatis 3.0.0-SNAPSHOT on JVM (powered by Quarkus 3.0.0.Final) started in 6.506s. Listening on: http://localhost:45125
2023-04-14 01:05:43,441 INFO  [io.quarkus] (main) Profile test activated. 
2023-04-14 01:05:43,441 INFO  [io.quarkus] (main) Installed features: [agroal, camel-core, camel-direct, camel-jta, camel-mock, camel-mybatis, cdi, jdbc-h2, mybatis, narayana-jta, resteasy, resteasy-jackson, smallrye-context-propagation, vertx]
2023-04-14 01:05:44,908 INFO  [org.apa.cam.com.moc.MockEndpoint] (executor-thread-1) Asserting: mock://results is satisfied
Error:  Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 16.452 s <<< FAILURE! - in org.apache.camel.quarkus.component.mybatis.it.MyBatisConsumerTest
Error:  testSelectList  Time elapsed: 3.481 s  <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
JSON path size() doesn't match.
Expected: <2>
  Actual: <1>
```
It needs to wait until the `onConsume` statments complete.

--- a/integration-tests/mybatis/pom.xml
+++ b/integration-tests/mybatis/pom.xml
@@ -59,6 +59,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-jdbc-h2</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.awaitility</groupId>
+            <artifactId>awaitility</artifactId>
+        </dependency>
 
         <!-- test dependencies -->
         <dependency>

--- a/integration-tests/mybatis/src/main/java/org/apache/camel/quarkus/component/mybatis/it/MybatisResource.java
+++ b/integration-tests/mybatis/src/main/java/org/apache/camel/quarkus/component/mybatis/it/MybatisResource.java
@@ -18,6 +18,7 @@ package org.apache.camel.quarkus.component.mybatis.it;
 
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
@@ -36,6 +37,9 @@ import org.apache.camel.CamelContext;
 import org.apache.camel.ProducerTemplate;
 import org.apache.camel.component.mock.MockEndpoint;
 import org.apache.camel.quarkus.component.mybatis.it.entity.Account;
+import org.hamcrest.Matchers;
+
+import static org.awaitility.Awaitility.await;
 
 @Path("/mybatis")
 @ApplicationScoped
@@ -136,6 +140,11 @@ public class MybatisResource {
         context.getRouteController().startRoute("mybatis-consumer");
         MockEndpoint.assertIsSatisfied(context);
 
-        return template.requestBody("mybatis:selectProcessedAccounts?statementType=SelectList", null, List.class);
+        List<?> body = await()
+                .atMost(1, TimeUnit.SECONDS)
+                .until(() -> template.requestBody("mybatis:selectProcessedAccounts?statementType=SelectList", null, List.class),
+                        Matchers.notNullValue());
+
+        return body;
     }
 }

